### DISCUSSION TITLE
Wifi Dropouts - Would that Cause Issues?

### DISCUSSION THREAD

**Discussion by Lunchtime0614** [doudar/SmartSpin2k#383] [Q&A]
For some reason my wifi is not able to hold a signal on the SS2k anymore - even though the access point is less than 5 feet from it.  I'm wondering though if the dropping of the wifi would also cause issues with the general use of the SS2k?  I'm seemingly having more issues now with the SS2k connecting, but not displaying power, or displaying power but not adjusting resistance....and it seems to be only when I have a wifi issue - a gut feeling that is.

**Comment by doudar** [comment]
Is it possible that the esp32 is getting hot?

**Comment by Lunchtime0614** [comment]
No. It does it as soon as it’s plugged in. That is starts dropping the
WiFi.

**Comment by doudar** [comment]
Barring a hardware issue, the only thing I could think of is if you're trying to use a 5ghz network. Those don't work well/at all.

You could also try updating to the latest firmware if you haven't.

Edit: It could be doing an automatic upgrade as well. If the blue light is solid or flickering, it's trying to update. That normally takes a couple minutes after boot.

**Comment by Lunchtime0614** [comment]
My network is both 2.4 and 5 with the same ssid. I’ll try adding it to my
guest network which is only 2.4.

**Comment by Lunchtime0614** [comment]
Well now I can't even get it to connect to wifi at all - didn't I see somewhere here that there is a way to reset the device completely?  

I'm wondering if the ESP32 wifi has failed.... @eMadman  sent me all the electronics and 3d parts since I didn't have a way to print them.  I'm thinking I may just need to build a whole new board - but every time I try to go source parts I find it difficult to determine if I'm getting the correct component (descriptions/specs are slightly different)....is there an updated parts list somewhere with USA links, or someone on here I can just buy all the electronics from their inventory (I also run into minimum order quantity issues) to build 2-3 additional boards?

**Reply by doudar** [reply]
You can reset to defaults by holding both shifters while powering it on.

**Reply by doudar** [reply]
I don't have a ton of inventory but I can send you a new board. 

Most of my efforts lately are on development of a new pcb with no soldering required. It *should* be available by the end of the quarter.

**Reply by eMadman** [reply]
Email me and I'll get you sorted out on whatever parts you need - the small components like resistors and caps tend to vary every re-order based on what's in stock that week.  I have spare blank PCBs and a handful of already tested esp32's ready to go.  Ordering everything locally, you'll need to use a combination of digikey (or similar), amazon, and aliexpress.  I find the amazon esp32's to have an extremely high failure rate (nearly 50% don't pass testing before I even solder the board!) while the cheaper ones on Aliexpress have been more reliable for me - less than 10% fail testing.


I would like to troubleshoot this further though.  Can you connect your SS2K to a PC and power it on with the flasher utility enabled?  I'm curious what comes up in the log view in the flasher.  Especially on first boot.  Quick way to restart it for logging purposes is to click the EN button on the esp32 (it's the one to the right when you're looking at the front of the case)

![image](https://user-images.githubusercontent.com/24726844/195095769-315f5172-3e68-488b-811e-284167ed6e0f.png)

Copy and paste the logs over here - it'll give an idea of what's happening during  boot.

Next thing worth trying is to flash the latest firmware using the flasher utility.  Step by step instructions are on the wiki: 
https://github.com/doudar/SmartSpin2k/wiki/Loading-Software


Last thing worth trying is to separate the 2.4ghz and 5ghz SSIDs on your wifi.  I find a lot of consumer access points and router don't do a graceful handoff between the two bands... or occasionally one of the bands enters into a power save mode so your phone and laptop think its all good when 2.4ghz might not be available for IoT devices.  Having the SSID separated helped massively for wifi stability in my condo

**Reply by Lunchtime0614** [reply]
@eMadman  - I'll do this testing today - with the flasher.  I'll also send you an email - I might just want to have spare parts on hand anyway....

**Reply by Lunchtime0614** [reply]
Here's the flasher when I boot - 
![image](https://user-images.githubusercontent.com/68966864/195114256-ef20ff8a-23c1-4ab4-ba62-0b3357c2dba5.png)

**Reply by eMadman** [reply]
This is good - it means it's booted up and bluetooth is working.  Can you click the "en" button? (to the right of the USB connector when you're looking at the module) and copy/paste the stream of text that shows immediately following that click?

It will reboot and initialize wifi - I want to see what the logs show when it's trying to connect to your wifi and whether or not the failsafe "smartspin2k" SSID is being created if a connection could not be established.

**Reply by Lunchtime0614** [reply]
Well I probably did things out of your preferred order.  I then reset it as @doudar  said and becuase I separated my wifi's band's into a separate 2.4 and 5 ghz and stopped it from trying to auto switch.  So now it's on the 2.4 ssid...

[10:15:28][179753][E](HTTP_Server): Rebooting fRequest
[10:15:30][181692][E](ERG_Mode): No data found in powertable.
[10:15:30][181692][E](ERG_ModeJ'rdata found in powertable.
[10:15:34][10:15:34][    35][E](Main): Mounting Filesystem
[10:15:34][    41][E](Config): Reading T /config.txt
[10:15:34][    70][E]i: Config File Loaded: /config.txt
[10:15:34][    71][E](Config): Contents of file: /config.txt
[10:15:34][   106][E](Config): Writing File: /config.txt
[10:15:34][   136][E](Config): Reading File: /userPWC.txt
[10:15:34][   167][E](Config): Config File Loaded: /userPWC.txt
[10:15:34][   16SUuConfig): Contents of file: /userPWC.txt
[10:15:34][   212][E](Config)'ɥѥFile: /userPWC.txt
[10:15:34][   259][E](Main):  read:ms=4d
[10:15:34][   291][E](Main):  read:current=13d
[10:15:34][   291][E](Main):  read:ms=4d
[10:15:34][   300][E](Main): Setting up cpu Tasks
[10:15:34][   300][E](HTTP_Server): Connecting to: xxxx
[10:15:35][  1460][E](HTTP_Server): Waiting for connection to be establZ...
[10:15:36][  2461][E](HTTP_ServerJ'ѥfor connection to be established...
[10:15:36][  2477][E](H5Uٕɥ: Connected to xxxx IP address: 192.168.145.153
[10:15:36][  2478][E](HTTP_Server): Open http://SmartSpin2K.local/+muHTTP_Server): Syncing clock...
[10:15:36][  2480][E](HTTP_Server): Clock synced to: 800
[10:15:36][  2483][E](HTTP_ٕɥ: Checking for newer firmware:
[10:15:41][  7590][E](HTTP_Server):   - Server version: 22.10.8
[10:15:41][  7625][E](HTTP_Server): New firmware detected!
[10:15:41][  7626][E](HTTP_Server): Upgrading from 22.9.5-develop to 22.10.8
[10:15:41][  7626][E](HTTP_Server): Updating FileSystem
[10:15:43][  8771][E](HTTP_Server): (Vѕ <?
[10:15:49][ 15206][E](HTTP_Server): Created: <?
[10:15:51][ 17035][E](HTTP_Server): Created: /favicon.ico
[10:15:52][ 18600WTBQQA}Mٕɥ: Created: \?
[10:15:53][ 19375][E](HTTP_Server): Created: /index.html
[10:15:56][ 22161][E](HTTP_Server): CreXWzRՕjs.gz
[10:15:57][ 23164][E](HTTP_Server): Created: <?
[10:15:58][ 24344][E](HTTP_Server): Created: /shift.html
[10:16:00][ 25932][E](HTTP_Server): Created: /status.html
[10:16:01][ 27217][E](HTTP_Server): Created: /style.css
[10:16:02][ 28224][E](HTTP_Server): Created: \?
[10:16:03][ 28869][E](HTTP_Server): Created: /develop.html
[10:16:03][ 28870][E](HTTP_Server): Updating Firmware...Please Wait

**Reply by eMadman** [reply]
Sweet!  A successful connection to wifi and it looks like the OTA update was also applied

Can you reach it at http://192.168.145.153 or http://smartspin2k.local?

**Reply by Lunchtime0614** [reply]
> Sweet! A successful connection to wifi and it looks like the OTA update was also applied
> 
> Can you reach it at http://192.168.145.153 or http://smartspin2k.local?

Yes - so far it's been holding a stable connection - as in the page doesn't drop or show unable to be loaded.  I'm going to leave the SS2K on today (I normally unplug it) and just randomly refresh the page to see if it's staying connected.  All yesterday it was hit or miss or would disconnect as it had a page half loaded.

**Reply by Lunchtime0614** [reply]
> > Sweet! A successful connection to wifi and it looks like the OTA update was also applied
> > Can you reach it at http://192.168.145.153 or http://smartspin2k.local?
> 
> Yes - so far it's been holding a stable connection - as in the page doesn't drop or show unable to be loaded. I'm going to leave the SS2K on today (I normally unplug it) and just randomly refresh the page to see if it's staying connected. All yesterday it was hit or miss or would disconnect as it had a page half loaded.

I may have spoke too soon - I just hit refresh on the page after typing the above and it was disconnected - although my router's admin page still showed it as connected - I have it set with a static IP as well.  I just restarted the SS2k and it reconnected.  I'll monitor more.

**Comment by Lunchtime0614** [comment]
Wifi is staying connected much better now - only was disconnected once every few hours that I saw...now it wasn't being used at the time either.

So a question for the dev's - does it go to sleep or disconnect from wifi after a certain amount of time of inactivity goes by?  Otherwise it seems to be staying connected much better than before.

**Reply by eMadman** [reply]
If it's disconnecting after a few hours, I think you're in good shape.  

I suspect your router may be treating it as an idle device and it's not gracefully returning it to an active state.  You can try enabling UDP logging in the settings - this will have it pumping out some data over wifi all the time, but even this will idle out after a while if the device is untouched for a few hours - if nothing's happening, there's nothing to log.  I think the recurring "No data found in powertable." from my screenshot below is sufficient since that updates in the log every few seconds.  This message will differ if you did a ride and a power table was generated.

![image](https://user-images.githubusercontent.com/24726844/195354953-49f748a7-214d-4bf5-aba7-0880980eed8b.png)

**Comment by doudar** [comment]
A new automatic update came out a few days ago. Please let me know if that fixed any of your issues.
